jerry-core/ecma/base/ecma-globals.h

```diff
@@ -121,6 +121,14 @@ typedef enum
   ECMA_PARSE_GENERATOR_FUNCTION = (1u << 8), /**< generator function is parsed */
 
   /* These flags are internally used by the parser. */
+#ifndef JERRY_NDEBUG
+  ECMA_PARSE_INTERNAL_FOR_IN_OFF_CONTEXT_ERROR = (1u << 9), /**< This flag is enabled when the for in/off statement
```

Comment:
Maybe shorter:
This flag represents an error in for in/of statements, which cannot be set if the parsing is completed successfully.

You can put the description before the declaration, or put the it before the flag is set in js-parser-statm.c.

Reply:
Thanks, updated.